### PR TITLE
bump(coreos/etcd): fix etcd hash in Godeps.json

### PR DESCRIPTION
**Pull request by ncdc**
Replacement for #9445 

@sjenning @smarterclayton

### CODE REVIEW

**Comment by ncdc** [comment]
[test]

File: hack/install-etcd.sh [hunk 1]
```diff
@@ -30,6 +30,11 @@ else
   pushd etcd >/dev/null
 fi
 
+export GOPATH=${PWD}/gopath
+
+rm -f $GOPATH/src/github.com/coreos/etcd
```
Comment (stevekuznetsov) [inline hack/install-etcd.sh]: Prefer `"${var}"` always, missing braces and quotes here

Comment (ncdc) [inline hack/install-etcd.sh]: Fixed. Also made ${PWD} vs $(pwd) consistent.

File: hack/install-etcd.sh [hunk 2]
```diff
@@ -30,11 +30,17 @@ else
...
+rm -f ${GOPATH}/src/github.com/coreos/etcd
+mkdir -p ${GOPATH}/src/github.com/coreos
+ln -s ${PWD} ${GOPATH}/src/github.com/coreos/etcd
+
 ./build
 
 echo
 echo Installed coreos/etcd ${etcd_version} into:
-echo export PATH=$(pwd):\$PATH
+echo export PATH=${PWD}/bin:\$PATH
```
Comment (stevekuznetsov) [inline hack/install-etcd.sh]: Also I will say `$(pwd)` and `"${PWD}"` are not identical -- IIRC `$(pwd)` will give you physical locations i.e. following symlinks whereas `"${PWD}"` will just print out the symlinks.

Comment (stevekuznetsov) [inline hack/install-etcd.sh]: (but I know this is a mostly useless nit on what looks like a debugging statement)

File: hack/install-etcd.sh [hunk 3]
```diff
@@ -30,11 +30,17 @@ else
   pushd etcd >/dev/null
 fi
 
+# setup a private GOPATH so the build can succeed
+export GOPATH=${PWD}/gopath
+rm -f ${GOPATH}/src/github.com/coreos/etcd
```
Comment (stevekuznetsov) [inline hack/install-etcd.sh]: @ncdc still would prefer to quote expansions - never know when someone's `pwd` has a space/newline/whatever in it

Comment (ncdc) [inline hack/install-etcd.sh]: sure thing

**Comment by stevekuznetsov** [comment]
Bash bits LGTM

**Comment by eparis** [comment]
[test] bot test!

**Comment by stevekuznetsov** [comment]
@eparis the poor bot can only take so much abuse :)

**Comment by ncdc** [comment]
[test]

**Comment by eparis** [comment]
[merge] !

**Pull request merged**